### PR TITLE
[refactor] Adding Context in core/ui to refetch data after transactions

### PR DESCRIPTION
**Pull request by vanminh1701**

### CODE REVIEW

**Comment by vanminh1701** [comment]
- we need wrap project with CandyProvider to use internal context
- I separate 2 contexts for CandyContext and CandyActionContext cuz when CandyContext change, the component use CandyActionContext will not affect (re-render), if the re-render is not a big deal, we can merge it to one store provider.
- setRefetch({}) to trigger render for all consumers
- current timeout=10s, need backend side handle update latest content, cuz I tested with 7s, it doesn't get latest content.

File: core/ui/src/public/Orders/index.tsx [hunk 1]
```diff
@@ -1,45 +1,13 @@
...
 import { CandyShop, OrderSortBy } from '@liqnft/candy-shop-sdk';
 import { InfiniteOrderList } from 'components/InfiniteOrderList';
 import { AnchorWallet } from '@solana/wallet-adapter-react';
-import './index.less';
+import { CandyContext } from 'public/Context';
 
-const ORDER_FETCH_LIMIT = 12;
-const LOADING_SKELETON_COUNT = 4;
-
-const SORT_OPTIONS: { value: OrderSortBy; label: string }[] = [
```
Comment (haVincy) [inline core/ui/src/public/Orders/index.tsx:13]: Why we're moving these constants to bottom..?  Generally, it should be at top in the common coding convention for programming languages? 

BTW, I feel we can create a `constant.ts` file in Orders folder for these datas that specific for `Orders/index.tsx`.

File: core/ui/src/public/Context/index.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+type CandyActionContextType = {
+  setRefetch: React.Dispatch<any>;
+};
+
+export const CandyActionContext = createContext<CandyActionContextType>({
+  setRefetch: () => console.log('no setRefetch')
+});
+
+interface CandyProp {
```
Comment (haVincy) [inline core/ui/src/public/Context/index.tsx:19]: `RevalidDataProp`

**Comment by haVincy** [comment]
> current timeout=10s, need backend side handle update latest content, cuz I tested with 7s, it doesn't get latest content.

@RustySol , any solution that backend can provide ? 

It looks not a good UX that user will have to wait 10s to refetch. What if user doing other actions that also trigger the refetch....

File: core/ui/src/public/Context/index.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import React, { createContext, ReactElement, useMemo, useState } from 'react';
+
+type CandyContextType = {
+  refetch: any;
```
[the remark this answers is not in the document]
Comment (vanminh1701) [reply core/ui/src/public/Context/index.tsx:4]: My target is when we update refetch and all consumers will re-render 1 time, so we have setRefetch({}) for that. But I will update setRefetch into a useReducer to no need pass {} anymore. 
will change to DataContextType

File: core/ui/src/public/Context/index.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+export const CandyActionContext = createContext<CandyActionContextType>({
+  setRefetch: () => console.log('no setRefetch')
+});
+
+interface CandyProp {
+  children: ReactElement;
+}
+
+export const CandyProvider: React.FC<CandyProp> = ({ children }) => {
```
[the remark this answers is not in the document]
Comment (vanminh1701) [reply core/ui/src/public/Context/index.tsx:23]: The user will import this CandyProvider to their project, so I think we keep it as a general name for use don't need to know what is the meaning of it

[the remark this answers is not in the document]
Comment (haVincy) [reply core/ui/src/public/Context/index.tsx:23]: okay, then can we do `CandyShopProvider`.

**Comment by RustySol** [comment]
> > current timeout=10s, need backend side handle update latest content, cuz I tested with 7s, it doesn't get latest content.
> 
> @RustySol , any solution that backend can provide ?
> 
> It looks not a good UX that user will have to wait 10s to refetch. What if user doing other actions that also trigger the refetch....

Lets do 3s for now, I think on mainnet it aint that slow

File: core/ui/src/components/BuyModal/index.tsx [hunk 5]
```diff
@@ -89,6 +93,10 @@ export const BuyModal: React.FC<BuyModalProps> = ({
       });
   };
 
+  useEffect(() => {
+    return () => clearTimeout(timeout);
+  }, []);
```
[the remark this answers is not in the document]
Comment (vanminh1701) [reply core/ui/src/components/BuyModal/index.tsx:98]: okay, great idea, I will call it useUnmountTimeout

File: core/ui/src/constant/index.ts [hunk 6]
```diff
@@ -0,0 +1 @@
+export const TIMEOUT_REFETCH_NFT = 3_000;
```
Comment (haVincy) [inline core/ui/src/constant/index.ts:1]: minor - let's add a comment to describe the real time updating issue as record and add TODO.

**Review by haVincy — APPROVED** [review]
Let's merge it fist but need to update Readme about the adding in Embedded UI Usages section.

**Pull request merged**